### PR TITLE
Batch fix for projections analysis with analyzer.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Interpreters/ActionsDAG.cpp [hunk 1]
```diff
@@ -796,11 +795,18 @@ ActionsDAGPtr ActionsDAG::foldActionsByProjection(const std::unordered_map<const
...
                     if (!node)
                     {
-                        bool should_rename = !rename.empty() && new_input->result_name != rename;
-                        const auto & input_name = should_rename ? rename : new_input->result_name;
-                        node = &dag->addInput(input_name, new_input->result_type);
-                        if (should_rename)
-                            node = &dag->addAlias(*node, new_input->result_name);
+                        auto & mapped_input = inputs_mapping[rename];
+
+                        if (!mapped_input)
```
Comment (KochetovNicolai) [inline src/Interpreters/ActionsDAG.cpp:800]: It is possible to have a few aliases on the same column. We may want to replace all the aliases, in this case they should have a single input as a child.

File: src/Interpreters/ActionsDAG.cpp [hunk 2]
```diff
@@ -836,7 +842,12 @@ ActionsDAGPtr ActionsDAG::foldActionsByProjection(const std::unordered_map<const
     }
 
     for (const auto * output : required_outputs)
-        dag->outputs.push_back(mapping[output]);
+    {
+        const auto * mapped_output = mapping[output];
```
Comment (KochetovNicolai) [inline src/Interpreters/ActionsDAG.cpp:846]: Keep the names for outputs.
Add an alias if the mapped node has a different result name.

File: src/Processors/QueryPlan/Optimizations/optimizeUseAggregateProjection.cpp [hunk 3]
```diff
@@ -144,9 +159,9 @@ bool areAggregatesMatch(
             /// and we can't replace one to another from projection.
             if (!candidate.function->getStateType()->equals(*aggregate.function->getStateType()))
             {
-                LOG_TRACE(&Poco::Logger::get("optimizeUseProjections"), "Cannot match agg func {} vs {} by state {} vs {}",
-                    aggregate.column_name, candidate.column_name,
-                    candidate.function->getStateType()->getName(), aggregate.function->getStateType()->getName());
+                // LOG_TRACE(&Poco::Logger::get("optimizeUseProjections"), "Cannot match agg func {} vs {} by state {} vs {}",
```
Comment (novikd) [inline src/Processors/QueryPlan/Optimizations/optimizeUseAggregateProjection.cpp:162]: Maybe remove it completely?

File: src/Processors/QueryPlan/Optimizations/optimizeUseAggregateProjection.cpp [hunk 4]
```diff
@@ -126,14 +135,20 @@ bool areAggregatesMatch(
             //     "Cannot match agg func {} by name {}",
             //     aggregate.column_name, aggregate.function->getName());
 
-            return false;
+            return {};
         }
 
+        size_t num_args = aggregate.argument_names.size();
+
+        DataTypes argumen_types;
```
Comment (novikd) [inline src/Processors/QueryPlan/Optimizations/optimizeUseAggregateProjection.cpp:143]:
```suggestion
        DataTypes argument_types;
```

**Pull request merged**